### PR TITLE
For Amazon Athena Data Source, change it to use odbc package instead of RODBC package.

### PR DESCRIPTION
**Pull request by hidekoji**
# Description

For Amazon Athena Data Source, change it to use odbc package instead of RODBC package.

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/system.R [hunk 1]
```diff
@@ -1347,15 +1356,19 @@ queryPostgres <- function(host, port, databaseName, username, password, numOfRow
 }
 
 #' @export
-queryAmazonAthena <- function(driver = "", region = "", authenticationType = "IAM Credentials", s3OutputLocation = "", user = "", password = "", additionalParams = "", query = "", numOfRows = 0, stringsAsFactors = FALSE, as.is = TRUE, ...){
-  if(!requireNamespace("RODBC")){stop("package RODBC must be installed.")}
+queryAmazonAthena <- function(driver = "", region = "", authenticationType = "IAM Credentials", s3OutputLocation = "", user = "", password = "", additionalParams = "", query = "", numOfRows = -1, stringsAsFactors = FALSE, as.is = TRUE, ...){
+  if(!requireNamespace("odbc")){stop("package RODBC must be installed.")}
   conn <- getAmazonAthenaConnection(driver = driver, region = region, authenticationType = authenticationType, s3OutputLocation = s3OutputLocation, user = user, password = password, additionalParams = additionalParams)
   tryCatch({
+    # For backwawrd compatibility, if 0 is passed as numOfRows, change it to -1.
```
Comment (hideaki) [inline R/system.R:1363]: Add some more explanation about what backword compatibility this was about for record.

Comment (hidekoji) [reply R/system.R:1363]: Added more comment.

File: R/system.R [hunk 2]
```diff
@@ -658,7 +658,16 @@ getAmazonAthenaConnection <- function(driver = "", region = "", authenticationTy
   if (is.null(conn)) {
     #https://www.rdocumentation.org/packages/RODBC/versions/1.3-15/topics/odbcConnect
     #For better performance, set max value (i.e. 1024) for rows_at_time
-    conn <- RODBC::odbcDriverConnect(connection = connectionString, rows_at_time = 1024)
+    conn <- DBI::dbConnect(
```
Comment (hideaki) [inline R/system.R:661]: Add comment on why we are still generating connection string above even though we don't pass it to this call.

Comment (hidekoji) [reply R/system.R:675]: Added comment.

**Pull request merged**